File: Java/02/tasks-manager-application/src/main/java/org/tasksmanager/Repositories/TaskTypesRepository.java
package org.tasksmanager.Repositories;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ConcurrentHashMap;
import javax.inject.Named;
import org.commons.Services.TaskTypeDao;
import org.cqrs.Repository;

@Named("taskTypesRepository")
public class TaskTypesRepository implements Repository<TaskTypeDao, String> {

    private static final ConcurrentHashMap<String, TaskTypeDao> _storage = new ConcurrentHashMap<String, TaskTypeDao>();

    @Override
    public void Delete(String id) {
        if (_storage.containsKey(id)) {
            _storage.remove(id);
        }
    }

    @Override
    public List<TaskTypeDao> GetAll() {
        return new ArrayList<>(_storage.values());
    }

    @Override
    public TaskTypeDao GetById(String id) {
        if (_storage.containsKey(id)) {
            return _storage.get(id);
        }
        return null;
    }

    @Override
    public TaskTypeDao Save(TaskTypeDao item) {
        _storage.put(item.getCode(), item);
        return item;
    }
}

File: Net/02/TasksManager.Application/Repositories/TaskTypesRepository.cs
using Commons.Services;
using Cqrs;
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;

namespace TasksManager.Repositories
{
    public class TaskTypesRepository : IRepository<TaskTypeDao, String>
    {
        private static ConcurrentDictionary<String, TaskTypeDao> _storage = new ConcurrentDictionary<String, TaskTypeDao>();

        public void Delete(String id)
        {
            TaskTypeDao deleted;
            if (_storage.ContainsKey(id)) _storage.TryRemove(id, out deleted);
        }

        public List<TaskTypeDao> GetAll()
        {
            return _storage.Values.ToList();
        }

        public TaskTypeDao GetById(String id)
        {
            if (_storage.ContainsKey(id)) return _storage[id];
            return null;
        }

        public TaskTypeDao Save(TaskTypeDao item)
        {
            _storage[item.Code] = item;
            return item;
        }
    }
}
